### PR TITLE
Improve sed detection in update-locales.sh

### PR DESCRIPTION
**Pull request by silverwind**
- ~~Make scripts work from any directory~~
- Detect sed version just like Makefile does

### CODE REVIEW

**Comment by silverwind** [comment]
Had to temporarily disable `-e` because BSD sed will exit non-zero for a `--version argument`, in fact it has no way to show its version:

```
$ /usr/bin/sed --version; echo $?
/usr/bin/sed: illegal option -- -
usage: sed script [-Ealnru] [-i extension] [file ...]
  sed [-Ealnu] [-i extension] [-e script] ... [-f script_file] ... [file ...]
1
```

**Comment by silverwind** [comment]
Removed the `cd` related changes. The scripts run under `/bin/sh` so bash variables were not there and I'm not 100% sure I can switch these scripts to bash because it's not present in alpine for example. So this is only the `sed` change now.

**Comment by wxiaoguang** [comment]
I didn't know where this script runs in. If it runs in the environment without bash (really? it's 2023 now ...), I think we should also revert the shebang line to `#!/bin/sh`.


ps: I didn't see today's crowdin sync commit, does  it mean that this script failed because I used `bash` in last PR?
I want to mention some maintainers but I don't know who knows the crwodin sync infrastructure ....

**Comment by lunny** [comment]
I think it's copied from `Makefile` and it works well before with no `gsed` installation requirement.

**Comment by silverwind** [comment]
Alpine is pretty much the only environment in use today that does not have `bash` unless manually installed. I also don't know where/how this script runs. Maybe @techknowlogick has some insight.

File: build/update-locales.sh [hunk 1]
```diff
@@ -1,13 +1,12 @@
 #!/bin/bash
```
Comment (wxiaoguang) [inline build/update-locales.sh]:
```suggestion
#!/bin/sh

# this script runs in alpine image which only has `sh` shell
```

Sorry, I made a mistake and I think it's incorrect to use bash in this script.

The last run fails: https://drone.gitea.io/go-gitea/gitea/68658/1/3

So we still need to use `sh`

**Comment by wxiaoguang** [comment]
I think we need to take this fix with the shebang line fix .... 😂

**Review by lunny — APPROVED** [review]

**Comment by silverwind** [comment]
As I suspected, this step is running on `alpine:3.17`, so it needs `sh` (busybox). I hope the script is sh-compatible.

**Comment by silverwind** [comment]
Need to merge this to un-break the translation updates.

**Review by jolheiser — APPROVED** [review]

**Pull request merged**